src/QueueReceiver.Core/Services/PersonService.cs

```diff
@@ -75,7 +75,8 @@ public async Task UnVoidPersonAsync(long personId)
 
             if (person?.Oid != null && await _graphService.AdPersonFoundInDeletedDirectory(person.Oid))
             {
-                return await CreatePerson(adPerson, shouldReconcile: true);
+                person.Oid = adPerson.Oid;
```

Comment:
One more thing, In `CreateIfNotExist` - the values in input parameter `userOid` and `adPerson.Oid` are used different places. They should have the same value, yes? To avoid confusion I would prefer that `userOid` is used consitently all places within the method.